internal/kafka/test/integration/admin_kafka_suspension_test.go

```diff
@@ -0,0 +1,144 @@
+			MultiAZ:               true,
+			Schedulable:           true,
+			KafkaInstanceLimit:    2,
+			Status:                api.ClusterWaitingForKasFleetShardOperator,
+			ProviderType:          api.ClusterProviderStandalone, // ensures there will be no errors with this test cluster not being available in ocm
+			SupportedInstanceType: api.AllInstanceTypeSupport.String(),
+		},
+	}
+	h, publicClient, teardown := kafkatest.NewKafkaHelperWithHooks(t, ocmServer, func(d *config.DataplaneClusterConfig) {
+		d.DataPlaneClusterScalingType = config.ManualScaling
+		d.ClusterConfig = config.NewClusterConfig(clusterList)
+	})
+	defer teardown()
+
+	// run test only on mock mode - fleetshard sync will always be mocked so there's no point running with real env.
+	ocmConfig := test.TestServices.OCMConfig
+	if ocmConfig.MockMode != ocm.MockModeEmulateServer || h.Env.Name == environments.TestingEnv {
+		t.SkipNow()
+	}
+
+	// run mock fleetshard sync
+	mockKasFleetshardSyncBuilder := kasfleetshardsync.NewMockKasFleetshardSyncBuilder(h, t)
+	mockKasFleetshardSync := mockKasFleetshardSyncBuilder.Build()
+	mockKasFleetshardSync.Start()
+	defer mockKasFleetshardSync.Stop()
+
+	// wait for data plane cluster to become `ready`
+	cluster, checkReadyErr := common.WaitForClusterStatus(test.TestServices.DBFactory, &test.TestServices.ClusterService, clusterList[0].ClusterId, api.ClusterReady)
+	g.Expect(checkReadyErr).NotTo(gomega.HaveOccurred(), "error waiting for data plane cluster to be ready: %s %v", cluster.ClusterID, checkReadyErr)
+
+	// setup private admin client
+	adminClientCtx := NewAuthenticatedContextForAdminEndpoints(h, []string{testFullRole})
+	adminClient := test.NewAdminPrivateAPIClient(h)
+
+	// setup Kafka instance
+	publicClientCtx := h.NewAuthenticatedContext(h.NewRandAccount(), nil)
+	kafkaRequestPayload := public.KafkaRequestPayload{
+		Region:        mocks.MockCluster.Region().ID(),
+		CloudProvider: mocks.MockCluster.CloudProvider().ID(),
+		Name:          mockKafkaName,
+		Plan:          fmt.Sprintf("%s.x1", types.STANDARD.String()),
+	}
+	publicKafkaReq, resp, err := publicClient.DefaultApi.CreateKafka(publicClientCtx, true, kafkaRequestPayload)
+	if resp != nil {
+		resp.Body.Close()
+	}
+	g.Expect(err).ToNot(gomega.HaveOccurred(), "failed to create a Kafka instance")
+
+	_, err = common.WaitForKafkaToReachStatus(publicClientCtx, kafkatest.TestServices.DBFactory, publicClient, publicKafkaReq.Id, kafkaconstants.KafkaRequestStatusPreparing)
+	g.Expect(err).ToNot(gomega.HaveOccurred(), "error waiting for kafka to reach status %q", kafkaconstants.KafkaRequestStatusPreparing)
+
+	// suspend Kafka instance
+	// set 'suspending: true' on a non-ready Kafka instance: should not update the status
+	suspendKafkaRequestPayload := private.KafkaUpdateRequest{
+		Suspended: &[]bool{true}[0],
+	}
+	privateKafkaReq, resp, err := adminClient.DefaultApi.UpdateKafkaById(adminClientCtx, publicKafkaReq.Id, suspendKafkaRequestPayload)
```

Comment:
I wonder if it makes sense to return an error if a user tries to suspend a Kafka from a non-ready state? Right now we accept it but do nothing. It may be a weird user experience as the request is successful but there would be no changes to the Kafka instance. 

wdyt @pawelpaszki @machi1990 @MikeEdgar?

Reply:
> It may be a weird user experience as the request is successful but there would be no changes to the Kafka instance.

+1, makes sense to return a 400 error.

Reply:
Thanks, I can probably add those changes in this PR. Unless it's preferable to have a separate PR for it?

Reply:
I think this PR is fine.

Reply:
+1 to reject the request